pytest.ini

```diff
@@ -5,9 +5,9 @@ faulthandler_timeout = 30
 markers =
     current: Mark tests during development
     imageformats: Require retrieving images from the web to test additional formats
-    exif: Require exif support
+    metadata: Require metadata support
```

Comment:
I will need more time with the tests, as I am currently unsure how to best set the markers and what to test where. As there is some train chaos (as usual :joy:) I will probably not make it today, and might end up pushing some commits instead of leaving a lot of comments for simplicity which you can cherry-pick, if you don't mind?

Note to self:
- update marker in tests, also end2end feature files
- think about `metadata` and `nometadata` markers, and definitely fixtures, it *should* be enough to mark the corresponding tests correctly, instead of forcing the code into a specific state via a fixture :thinking: This would also solve the version part, only test for a specific string if it is marked accordingly. Alternatively mocking could make sense as well. Processing :blush:

Reply:
This makes sense. Feel free to push to this branch.

I am glad you can help with sorting out the setup. Once this is done I will continue with fixing, and potentially adding, tests. :blush:

And I wish you a good journey! :steam_locomotive:

Reply:
So I pushed some stuff :blush: not super happy with the state, but at least there is some progress. I believe having the metadata test as integration instead of unit test makes sense, as we then don't have to go through the havoc of mocking, but can test the integration of `imutils.metadata` with the plugins. In principle, if we have an image with metadata, we could even check the actual implementation of `get_keys` and so forth methods in a semi-isolated manner.

The integration tests unfortunately still require both marking and using the corresponding fixture, in case we test piexif / pyexiv2 explicitly as I couldn't figure out how to mark a fixture (instead of the tests) quickly. Could definitely be improved.

Let me know if you have any thoughts on improving this :blush:

Note that there are still two failing test cases. Is there a reason the `MetadataHandler` does not raise for `get_keys` and `get_metadata`?

Reply:
Soo, you killed my commits? :joy: that is why I don't like pushing to other repositories ...

Reply:
Umm, that is not me force pushing. Let me investigate...

Reply:
That's a bit creepy then, maybe some github "protection" mechanism, I will just push into the `core/metadataplugin-jcjgraf` branch and you can grab the commits :blush:

Reply:
I think that has been resolved. I updated the PR description/todo to remember that we potentially need to improve the marking of fixtures